Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -414,10 +415,16 @@ - (BOOL)containsStatusCode:(NSInteger)statusCode
 
 - (void)addBreadcrumbForSessionTask:(NSURLSessionTask *)sessionTask
 {
-    if (!self.isNetworkBreadcrumbEnabled) {
+    id hasBreadcrumb
+        = objc_getAssociatedObject(sessionTask, &SENTRY_NETWORK_REQUEST_TRACKER_BREADCRUMB);
+    if (hasBreadcrumb && [hasBreadcrumb isKindOfClass:NSNumber.class] &&
+        [hasBreadcrumb boolValue]) {
         return;
     }
 
+    if (!self.isNetworkBreadcrumbEnabled) {
+        return;
+    }
```

Comment:
`m`: I would keep the check `!self.isNetworkBreadcrumbEnabled` before using `objc_getAssociatedObject` to avoid unnecessary work when `isNetworkBreadcrumbEnabled` is disabled.